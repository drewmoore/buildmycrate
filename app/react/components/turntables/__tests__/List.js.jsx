/* globals describe test expect */
import React          from 'react';
import { Provider }   from 'react-redux';
import renderer       from 'react-test-renderer';
import { mount }      from 'enzyme';
import TurntablesList from '../List.js.jsx';
import store          from '../../../__tests__/helpers/store.js.es6';
import '../../../__tests__/helpers/initialize.js.es6';

describe('TurntablesList', () => {
  const props    = { turntables: [{ id: 0, trackId: null }, { id: 1, trackId: null }] };
  const template = (<Provider store={store}><TurntablesList {...props} /></Provider>);

  describe('DOM Snapshot', () => {
    const tree = renderer.create(template).toJSON();

    test('should match snapshot', () => {
      expect(tree).toMatchSnapshot();
    });
  });

  describe('Rendering Props', () => {
    const icon = mount(template);

    test('should render components for each turntable', () => {
      const elements = icon.find('[data-hook="turntable-container"]');
      expect(elements).toHaveLength(props.turntables.length);
    });
  });
});
