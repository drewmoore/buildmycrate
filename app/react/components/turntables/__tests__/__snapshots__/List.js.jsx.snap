// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TurntablesList DOM Snapshot should match snapshot 1`] = `
<div
  className="row"
>
  <div
    className="col-xs-12 col-lg-offset-1 col-lg-10"
  >
    <div
      className="row"
    >
      <div
        className="col-xs-12 col-md-6"
      >
        <div
          className="row turntable-container alpha"
          data-hook="turntable-container"
        >
          <div
            className="col-xs-12 section console"
          >
            <div
              className="row"
            >
              <div
                className="col-xs-3"
              >
                <img
                  alt="Track"
                  className="turntable-track-image"
                  src={undefined}
                />
              </div>
              <div
                className="col-xs-9"
              >
                <div
                  className="row"
                >
                  <div
                    className="col-xs-9 fat-bottom"
                  >
                    <span />
                  </div>
                  <div
                    className="col-xs-3 fat-bottom text-right"
                  >
                    <span>
                      
                    </span>
                  </div>
                </div>
                <div
                  className="row"
                >
                  <div
                    className="col-xs-9"
                  >
                    <span>
                      
                    </span>
                  </div>
                  <div
                    className="col-xs-3 text-right"
                  >
                    <span>
                      
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="row"
            >
              <div
                className="col-xs-12 text-right"
              />
            </div>
            <div
              className="row"
            >
              <div
                className="col-xs-12 waveform-container"
                id="waveform-container-for-0"
              />
            </div>
            <div
              className="row"
            >
              <div
                className="col-xs-10 controls"
              />
              <div
                className="col-xs-2 controls text-center"
              >
                <img
                  alt="Loading Indicator"
                  className="spinner  "
                  src={undefined}
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="col-xs-12 col-md-6"
      >
        <div
          className="row turntable-container omega"
          data-hook="turntable-container"
        >
          <div
            className="col-xs-12 section console"
          >
            <div
              className="row"
            >
              <div
                className="col-xs-3"
              >
                <img
                  alt="Track"
                  className="turntable-track-image"
                  src={undefined}
                />
              </div>
              <div
                className="col-xs-9"
              >
                <div
                  className="row"
                >
                  <div
                    className="col-xs-9 fat-bottom"
                  >
                    <span />
                  </div>
                  <div
                    className="col-xs-3 fat-bottom text-right"
                  >
                    <span>
                      
                    </span>
                  </div>
                </div>
                <div
                  className="row"
                >
                  <div
                    className="col-xs-9"
                  >
                    <span>
                      
                    </span>
                  </div>
                  <div
                    className="col-xs-3 text-right"
                  >
                    <span>
                      
                    </span>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="row"
            >
              <div
                className="col-xs-12 text-right"
              />
            </div>
            <div
              className="row"
            >
              <div
                className="col-xs-12 waveform-container"
                id="waveform-container-for-1"
              />
            </div>
            <div
              className="row"
            >
              <div
                className="col-xs-10 controls"
              />
              <div
                className="col-xs-2 controls text-center"
              >
                <img
                  alt="Loading Indicator"
                  className="spinner  "
                  src={undefined}
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
